hydra-cardano-api/src/Hydra/Cardano/Api.hs

```diff
@@ -1,68 +1,546 @@
+  PoolId,
+  ProtocolParameters (..),
+  ShelleyGenesis (..),
+  ShelleyLedgerEra,
+  VerificationKey (..),
+  fromPlutusData,
+  toPlutusData,
+ )
+import Cardano.Api.UTxO (
+  UTxO,
+  UTxO' (..),
+ )
 import Hydra.Cardano.Api.Prelude (
   Era,
   LedgerEra,
   StandardCrypto,
  )
 
-import Cardano.Api hiding (UTxO, toLedgerUTxO)
-import Cardano.Api.Byron hiding (UTxO, toLedgerUTxO)
-import Cardano.Api.Shelley hiding (UTxO, toLedgerUTxO)
-import Cardano.Api.UTxO (UTxO, UTxO' (..))
-
-import Hydra.Cardano.Api.AddressInEra
-import Hydra.Cardano.Api.CtxTx
-import Hydra.Cardano.Api.CtxUTxO
-import Hydra.Cardano.Api.Hash
-import Hydra.Cardano.Api.KeyWitness
-import Hydra.Cardano.Api.Lovelace
-import Hydra.Cardano.Api.PlutusScript
-import Hydra.Cardano.Api.PlutusScriptVersion
-import Hydra.Cardano.Api.ScriptData
-import Hydra.Cardano.Api.ScriptDatum
-import Hydra.Cardano.Api.ScriptHash
-import Hydra.Cardano.Api.ScriptWitnessInCtx
-import Hydra.Cardano.Api.Tx
-import Hydra.Cardano.Api.TxBody
-import Hydra.Cardano.Api.TxId
-import Hydra.Cardano.Api.TxIn
-import Hydra.Cardano.Api.TxOut
-import Hydra.Cardano.Api.TxOutDatum
-import Hydra.Cardano.Api.TxOutValue
-import Hydra.Cardano.Api.TxScriptValidity
-import Hydra.Cardano.Api.UTxO
-import Hydra.Cardano.Api.Value
-import Hydra.Cardano.Api.Witness
+import Hydra.Cardano.Api.AddressInEra as X
+import Hydra.Cardano.Api.CtxTx as X
+import Hydra.Cardano.Api.CtxUTxO as X
+import Hydra.Cardano.Api.ExecutionUnits as X
+import Hydra.Cardano.Api.Hash as X
+import Hydra.Cardano.Api.KeyWitness as X
+import Hydra.Cardano.Api.Lovelace as X
+import Hydra.Cardano.Api.PlutusScript as X
+import Hydra.Cardano.Api.PlutusScriptVersion as X
+import Hydra.Cardano.Api.ScriptData as X
+import Hydra.Cardano.Api.ScriptDatum as X
+import Hydra.Cardano.Api.ScriptHash as X
+import Hydra.Cardano.Api.ScriptWitnessInCtx as X
+import Hydra.Cardano.Api.Tx as X
+import Hydra.Cardano.Api.TxBody as X
+import Hydra.Cardano.Api.TxId as X
+import Hydra.Cardano.Api.TxIn as X
+import Hydra.Cardano.Api.TxOut as X
+import Hydra.Cardano.Api.TxOutDatum as X
+import Hydra.Cardano.Api.TxOutValue as X
+import Hydra.Cardano.Api.TxScriptValidity as X
+import Hydra.Cardano.Api.UTxO as X
+import Hydra.Cardano.Api.Value as X
+import Hydra.Cardano.Api.Witness as X
+
+import qualified Cardano.Api
+import qualified Cardano.Api.Shelley
+import qualified Cardano.Ledger.Alonzo.Data as Ledger
+import qualified Cardano.Ledger.Alonzo.Scripts as Ledger
+import qualified Cardano.Ledger.Alonzo.TxBody as Ledger
+import qualified Cardano.Ledger.Alonzo.TxWitness as Ledger
+import qualified Cardano.Ledger.Keys as Ledger
+import qualified Cardano.Ledger.Shelley.Address.Bootstrap as Ledger
+import qualified Cardano.Ledger.Shelley.Tx as Ledger hiding (TxBody)
+
+-- NOTE: We always use the latest era available in our codebase, so to ease type
+-- signatures and notations, we specialize any type of the cardano-api normally
+-- parameterized by an era to the latest era 'Era'.
```

Comment:
Would be nice to have this as a module haddock a bit further up :)

Reply:
It's probably also worth that we strip away the `SomethingInEra` witnesses and the intermediary types like `TxOutValue` using the patterns in this module. Parameterized and not-stripped-off functions would be available in the individual `Hydra.Cardano.Api.XX` modules

Reply:
That is, maybe we want to move the explantions in the readme into the haddock and guide README readers there?